src/SdkCommon/ClientRuntime/ClientRuntime/Serialization/EnumConverterBase.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Reflection;
+using Newtonsoft.Json;
+
+namespace Microsoft.Rest.ClientRuntime.Serialization
+{
+    /// <summary>
+    /// Base class for custom JsonConverters.
+    /// </summary>
+    public abstract class EnumConverterBase : JsonConverter
```

Comment:
Why the name change? There is nothing about this class that is specific to the enum converter.

Also, after staring at this with fresh eyes, I think it would have been better if I had not used inheritance for re-use purposes here. These methods could just as easily be extensions on `JsonReader`. I leave it up to you to decide what fits best with the `ClientRuntime`.

Reply:
I was trying to semantically narrow the scope of its usage since a lot of common serialization related logic lies in this project and I did not want to introduce something new.
On similar lines if we do model these methods as extension methods for the JsonReader I wonder if it may cause a breaking change (I can't think of a reason why but just want to point it out)

@shahabhijeet @fearthecowboy @olydis I'd like your thoughts too

Reply:
@dsgouda Regarding backward compatibility, if you decide to make these extension methods of `JsonReader`, I would recommend doing so in an internal class so it is only visible within the ClientRuntime.